jung-algorithms/src/main/java/edu/uci/ics/jung/algorithms/util/IterativeContext.java

```diff
@@ -9,11 +9,44 @@
  */
 package edu.uci.ics.jung.algorithms.util;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 /** An interface for algorithms that proceed iteratively. */
 public interface IterativeContext {
+
+  interface WithPreRelax extends IterativeContext {}
```

Comment:
Adding a tagging subinterface to this interface seems a bit awkward to me.  As far as I can tell, the only time that this interface is ever implemented is by the `LayoutAlgorithm` abstract implementation that is an `IterativeContext`, so it seems that it should be sufficient to ask whether something is both a `LayoutAlgorithm` and an `IterativeContext`.   (We might be better served by creating a subinterface of `LayoutAlgorithm` that extends `IterativeContext` and checking for that.)

Do you anticipate that we might ever have a `LayoutAlgorithm` that would extend `IterativeContext` but _not_ want pre-relaxing?  Or that we might want some other (non-layout-related) `IterativeContext` that we'd want to tag as `WithPreRelax`?

If we do retain this interface, we need to add documentation that specifies the context(s) in which it should be used.